modin/core/io/text/text_file_dispatcher.py

```diff
@@ -630,3 +685,380 @@ def _validate_usecols_arg(cls, usecols):
+        -------
+        new_query_compiler : BaseQueryCompiler
+            New query compiler, created from `new_frame`.
+        """
+        new_index, row_lengths = cls._define_index(index_ids, index_name)
+        # Compute dtypes by collecting and combining all of the partition dtypes. The
+        # reported dtypes from differing rows can be different based on the inference in
+        # the limited data seen by each worker. We use pandas to compute the exact dtype
+        # over the whole column for each column. The index is set below.
+        dtypes = cls.get_dtypes(dtypes_ids) if len(dtypes_ids) > 0 else None
+        # Compose modin partitions from `partition_ids`
+        partition_ids = cls.build_partition(partition_ids, row_lengths, column_widths)
+
+        # Set the index for the dtypes to the column names
+        if isinstance(dtypes, pandas.Series):
+            dtypes.index = column_names
+        else:
+            dtypes = pandas.Series(dtypes, index=column_names)
+
+        new_frame = cls.frame_cls(
+            partition_ids,
+            new_index,
+            column_names,
+            row_lengths,
+            column_widths,
+            dtypes=dtypes,
+        )
+        new_query_compiler = cls.query_compiler_cls(new_frame)
+        skipfooter = kwargs.get("skipfooter", None)
+        if skipfooter:
+            new_query_compiler = new_query_compiler.drop(
+                new_query_compiler.index[-skipfooter:]
+            )
+        if skiprows_md is not None:
+            # skip rows that passed as array or callable
+            nrows = kwargs.get("nrows", None)
+            index_range = pandas.RangeIndex(len(new_query_compiler.index))
+            if is_list_like(skiprows_md):
+                new_query_compiler = new_query_compiler.view(
+                    index=index_range.delete(skiprows_md)
+                )
+            elif callable(skiprows_md):
+                mod_index = skiprows_md(index_range)
+                if not isinstance(mod_index, np.ndarray):
+                    mod_index = mod_index.to_numpy("bool")
+                view_idx = index_range[~mod_index]
+                new_query_compiler = new_query_compiler.view(index=view_idx)
+            else:
+                raise TypeError(
+                    f"Not acceptable type of `skiprows` parameter: {type(skiprows_md)}"
+                )
+
+            if not isinstance(new_query_compiler.index, pandas.MultiIndex):
+                new_query_compiler = new_query_compiler.reset_index(drop=True)
+
+            if nrows:
+                new_query_compiler = new_query_compiler.view(
+                    pandas.RangeIndex(len(new_query_compiler.index))[:nrows]
+                )
+        if index_col is None:
+            new_query_compiler._modin_frame.synchronize_labels(axis=0)
+
+        return new_query_compiler
+
+    @classmethod
+    def _generic_read(cls, filepath_or_buffer, callback, is_for_fwf_reader, **kwargs):
```

Comment:
A few comments here:

- Ideally this would be `_read` and the parameters like `callback` would be fields of the class. 
- Why are we handling `is_for_fwf_reader` like this? I don't like passing a boolean flag for special casing the code in this way. I would prefer to just override it (even if it's more code).
- Can we add type hints here?